### DISCUSSION TITLE
Accudraw compass on gltf

### DISCUSSION THREAD

**Discussion by Pravendra-rana** [iTwin/itwinjs-core#5338] [Q&A]
i was using accudraw and even after enabling smartRotation i am unable to see accuDraw compass

**Comment by bbastings** [comment]
Please provide more details. What are you trying to do and how did you try to achieve it (show code)?

**Reply by Pravendra-rana** [reply]
![for_simple3d](https://user-images.githubusercontent.com/117334414/229509238-d3e0d585-84bd-4e6e-be1c-58c3a98702f1.png)
![for_Gltf](https://user-images.githubusercontent.com/117334414/229509517-a617e948-14ec-4aac-97de-0f68d358f0c3.png

same tool used for both the decorators but gltf dacorator not showing the compass

**Reply by bbastings** [reply]
I'm going to need more than png files to try to help. These are pickable decorations, and you've written a tool that locates them? Show your code or create a sandbox to demonstrate the problem...

**Reply by Kshitij-Jadhav** [reply]
export class MeasurePointsTool extends PrimitiveTool {
  public static toolId = "Measure.Points";  
  public override async onPostInstall() {
    await super.onPostInstall();
    this.setupAndPromptForNextAction();
  }

 

  public setupAndPromptForNextAction(): void {
    IModelApp.accuSnap.enableSnap(true);

 

    if (0 === this.points.length) {
      IModelApp.notifications.outputPromptByKey(
        "SampleApp:tools.Measure.ByPoints.Prompts.FirstPoint"
      );
      return;
    }

 

    IModelApp.notifications.outputPromptByKey(
      "SampleApp:tools.Measure.ByPoints.Prompts.NextPoint"
    );

 

    const hints = new AccuDrawHintBuilder();
    hints.enableSmartRotation = true;

 

    if (
      this.points.length > 1 &&
      !this.points[this.points.length - 1].isAlmostEqual(
        this.points[this.points.length - 2]
      )
    )
      hints.setXAxis(
        Vector3d.createStartEnd(
          this.points[this.points.length - 2],
          this.points[this.points.length - 1]
        )
      ); // Rotate AccuDraw to last segment...

 

    hints.setOrigin(this.points[this.points.length - 1]);
    hints.sendHints();
  }
};

this is a part of code from the tool we are using for both SimpleGeometryDecorator and GltfDecorator

**Reply by bbastings** [reply]
Link to GltfDecorator? I assume this is being created as a pickable decoration? Is your decorator's "testDecorationHit" method being called when it's under the cursor? If your decorator isn't being located (you should see it flash in the hilite color when it's under the cursor) then it won't be used to adjust the point for the button event.

**Reply by tusharsable98** [reply]
Hi @bbastings , the decortor is pickable and  does highlight when cursor is moved upon. but the tools which we are using snap and show accudraw compass only for selecting the first point. once selected the compass is nowhere to be seen and we can just select arbitrary points  under the cross.
